### DISCUSSION TITLE
Aptos CLI E2E Testing Suite Connection Timeout

### DISCUSSION THREAD

**Discussion by ganymedio** [aptos-labs/aptos-developer-discussions#288] [General]
Anyone familiar with the e2e testing suite in `aptos-core/crates/aptos`?

when i run 

`poetry run python main.py --base-network mainnet --test-cli-tag nightly`

it gets to 

`Waiting for node and faucet APIs for aptos-tools-testnet to come up`

and then 

`ERROR - Timeout while waiting for node / faucet to come up`

and if i try

`poetry run python main.py -d --base-network mainnet --test-cli-path ~/aptos-core/target/debug/aptos`

it does 

`DEBUG - Starting new HTTP connection (1): 127.0.0.1:8080` 

over and over again and then gets a timeout.

Is there some step i'm missing? 

Thank you!

**Comment by gregnazario** [comment]
Hmm, mine also didn't start up correctly here.

I do know that sometimes the debug build is much slower than expected, and may need to instead build a local CLI with

```
cargo build -p aptos --profile cli
```

Then use `~/aptos-core/target/cli/aptos`, which will be faster to startup.

I would also ensure you have docker installed properly, and that there are no port conflicts with anything else running on your machine.

cc @banool
